Project21/Project21/Traingit.c

```diff
@@ -1,12 +1,144 @@
+   int b0[] = { -5,12,19,20,21,30 }, b1[] = { 4,8,54,55 }, b2[] = { -5,-3,2 }, b3[] = { 0,1,2,3,4 }, b4[] = { -4,-3,-2,-1 }, b5[] = { -250,15,17,46,60 },
+      b6[] = { -287,-67,78,122,223,2000 }, b7[] = { 2,3,4,5 },
+      * Expected[] = { b0,b1,b2,b3,b4,b5,b6,b7 };
+   for (int i = 0; i < 8; i++) {
+      printf ("\n---------------------------------------------------------------------------------------------");
+      printf ("\nInputArray    :   ");
+      for (int j = 0; j < size[i]; j++) printf ("%d ", jagged[i][j]);
+      BubbleSort (jagged[i], size[i]);
+      printf ("\nSorted Array  :  ");
+      for (int j = 0; j < size[i]; j++) printf ("%d ", jagged[i][j]);
+      printf ("\nExpected Array:  ");
+      for (int j = 0; j < size[i]; j++) printf ("%d ", Expected[i][j]);
+      printf ("\nBubble Sort   :   ");
+      int k = 0;
+      for (; k < size[i]; k++) {
+         if (jagged[i][k] != Expected[i][k])
+            break;
+      }
+      k == size[i] ? printf ("\033[32mPASS\033[0m") : printf ("\033[31mFAIL\033[0m");
+      if (k == size[i] - 1) printf ("\033[32mPASS\033[0m");
+   }
+   printf ("\n---------------------------------------------------------------------------------------------");
+}
+
+/// <summary>Function to run the test cases </summary>
+static void TestCases () {
+   int r0[] = { -5,12,19,20,21,30 }, r1[] = { 4,8,54,55 }, r2[] = { -5,-3,2 }, r3[] = { 0,1,2,3,4 },
+      r4[] = { -4,-3,-2,-1 }, r5[] = { -250,15,17,46,60 }, r6[] = { -287,-67,78,122,223,2000 }, r7[] = { 2,3,4,5 },
+      * jagged[] = { r0,r1,r2,r3,r4,r5,r6,r7 },
+      size[] = { 6,4,3,5,4,5,6,4 },
+      iptarget[] = { 20,54,2,3,-2,17,78,5 },
+      expectedArray[] = { 3,2,2,3,2,2,2,3 };
+
+   for (int i = 0; i < 8; i++) size[i];
+   printf ("\033[36m\n*****TestCases*****\n\n**********Binary Search**********\033[0m");
+   for (int i = 0; i < 8; i++) {
+      printf ("\n---------------------------------------------------------------------------------------------");
+      BubbleSort (jagged[i], size[i]);
+      int index = BinarySearch (jagged[i], size[i], iptarget[i]);
+      printf ("\nKey          :   %d\nexpectedArray:   %d\nIndex        :   %d\nBinary Search: ", iptarget[i], expectedArray[i], index);
+      printf ((index == expectedArray[i]) ? "\033[32mPASS\033[0m" : "\033[31mFAIL\033[0m");
+   }
+   printf ("\n---------------------------------------------------------------------------------------------");
+   printf ("\n");
+}
+
+int main () {
+   SortTestCases ();
+   TestCases ();
+   for (;;) {
+      printf ("\n*****CHOOSE*****\n1 = SortAndSearch\n2 = ClearScreen\n3 = Exit\nChoose any option:");
+      switch (_getch ()) {
+      case '1':
+         ClrScreen ();
+         SortAndSearch ();
+         break;
+      case '2':
+         ClrScreen ();
+         break;
+      case '3':
+         ClrScreen ();
+         return 0;
+      default:
+         printf ("\n\033[31mEnter a number between 1-4\n\033[0m");
+      }
+   }
+   return 0;
+}
```

Comment:
![image](https://github.com/user-attachments/assets/0e46bf07-7a11-4fdc-975a-23f8c7b40a73)
The output should be like this